test/hotspot/jtreg/runtime/Nestmates/protectionDomain/TestDifferentProtectionDomains.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8261395
+ * @summary Test the code paths when a nest-host and nest-member class are in
+ *          different protection domains and the compiler thread needs to
+ *          perform a nestmate access check.
+ * @comment We use WB to force-compile a constructor to recreate the original
+ *          failure scenario, so only run when we have "normal" compiler flags.
+ * @requires vm.compMode=="Xmixed" &
+ *           (vm.opt.TieredStopAtLevel == null | vm.opt.TieredStopAtLevel == 4)
+ * @library /test/lib /
+ * @build sun.hotspot.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller sun.hotspot.WhiteBox
+ * @compile Host.java
+ * @run main/othervm -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI -Xbootclasspath/a:.
+ *                   -Xlog:class+nestmates=trace,protectiondomain=trace
+ *                   TestDifferentProtectionDomains
+ */
+
+import java.lang.reflect.Constructor;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.Path;
+import java.security.ProtectionDomain;
+
+import compiler.whitebox.CompilerWhiteBoxTest;
+import sun.hotspot.WhiteBox;
+
+public class TestDifferentProtectionDomains {
+
+    static final String TARGET = "Host";
+
+    // We need a custom classloader so that we can
+    // use a different protection domain for our target classes.
+
+    static class CustomLoader extends ClassLoader {
+
+        static final String CLASSES = System.getProperty("test.classes");
+        static final Path CLASSES_DIR = Paths.get(CLASSES);
+
+        CustomLoader(ClassLoader parent) {
+            super(parent);
+        }
+
+        static byte[] getBytesForClass(String name) throws IOException {
```

Comment:
There's a ClassUnloadCommon library call for this.

Reply:
Thanks. I copied this from another test. If there's an existing utility I will switch to it.